src/main/java/sirius/db/mixing/EntityDescriptor.java

```diff
@@ -826,7 +826,9 @@ public Entity readFrom(String alias, Set<String> columns, ResultSet rs) throws E
     private void readIdAndVersion(String alias, Set<String> columns, ResultSet rs, Entity entity) throws SQLException {
         String idColumnLabel = getIdColumnLabel(alias);
         if (columns.contains(idColumnLabel.toUpperCase())) {
-            entity.setId(rs.getLong(idColumnLabel));
+            if (rs.getString(idColumnLabel) != null) {
+                entity.setId(rs.getLong(idColumnLabel));
```

Comment:
nope, use rs.wasNull() - *horror API* 

http://stackoverflow.com/questions/2920364/checking-for-a-null-int-value-from-a-java-resultset